week2/intro_to_property_packages_verbose.ipynb

```diff
@@ -0,0 +1,677 @@
+    "    m2.fs.feed.initialize()\n",
+    "    results = solver.solve(m2)\n",
+    "    print(f\"Solve termination {results.solver.termination_condition}\")\n",
+    "    # Unfix NaCl mass flow\n",
+    "    m2.fs.feed.properties[0].flow_mass_phase_comp[\"Liq\", \"NaCl\"].unfix()\n",
+    "\n",
+    "    return m2\n",
+    "\n",
+    "\n",
+    "m2 = build_NaCl_prop_model()\n",
+    "\n",
+    "# =======================================================================================\n",
+    "# Define a range of concentrations\n",
+    "low_concs = np.linspace(5, 70, 25)  # g/L\n",
+    "# Create empty lists to store osmotic pressures for each model\n",
+    "p_osm1_low = []\n",
+    "p_osm2_low = []\n",
+    "\n",
+    "\n",
+    "for c in low_concs:\n",
+    "    m1.fs.feed.properties[0].conc_mass_phase_comp[\"Liq\", \"TDS\"].fix(c)\n",
+    "    m2.fs.feed.properties[0].conc_mass_phase_comp[\"Liq\", \"NaCl\"].fix(c)\n",
+    "    results1 = solver.solve(m1)\n",
+    "    results2 = solver.solve(m2)\n",
+    "    p_osm1_low.append(\n",
+    "        value(\n",
+    "            pyunits.convert(\n",
+    "                m1.fs.feed.properties[0].pressure_osm_phase[\"Liq\"], to_units=pyunits.bar\n",
+    "            )\n",
+    "        )\n",
+    "    )\n",
+    "    p_osm2_low.append(\n",
+    "        value(\n",
+    "            pyunits.convert(\n",
+    "                m2.fs.feed.properties[0].pressure_osm_phase[\"Liq\"], to_units=pyunits.bar\n",
+    "            )\n",
+    "        )\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "# =======================================================================================\n",
+    "# Plot the results\n",
+    "\n",
+    "fig, ax = plt.subplots()\n",
+    "\n",
+    "ax.scatter(low_concs, p_osm1_low, color=\"blue\", label=\"Seawater Property Package\")\n",
+    "ax.scatter(low_concs, p_osm2_low, color=\"red\", label=\"NaCl Property Package\")\n",
+    "ax.set_xlabel(\"Concentration (g/L)\")\n",
+    "ax.set_ylabel(\"Osmotic Pressure (bar)\")\n",
+    "ax.set_title(\"Osmotic Pressure vs Concentration (Low Range)\")\n",
+    "ax.legend()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "73049798",
+   "metadata": {},
+   "source": [
+    "### Compare seawater and NaCl property packages\n",
+    "\n",
+    "Here, we repeat the same process as above but for high salinity (70-250 g/L)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "47ecf7ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m1 = build_seawater_prop_model()\n",
+    "m2 = build_NaCl_prop_model()\n",
+    "\n",
+    "# =======================================================================================\n",
+    "# Define a range of concentrations\n",
+    "high_concs = np.linspace(70, 250, 25)  # g/L\n",
+    "# Create empty lists to store osmotic pressures for each model\n",
+    "p_osm1_high = []\n",
+    "p_osm2_high = []\n",
+    "\n",
+    "for c in high_concs:\n",
```

Comment:
Consider adding an in-line comment above for-loop saying something like: solves for osmotic pressures across the range of concentrations?

Should also include in skinny version